distributed/client.py

```diff
@@ -786,13 +786,16 @@ def __repr__(self):
             return "<%s: not connected>" % (self.__class__.__name__,)
 
     def _repr_html_(self):
-        if (
-            self.cluster
-            and hasattr(self.cluster, "scheduler")
-            and self.cluster.scheduler
-        ):
-            info = self.cluster.scheduler.identity()
-            scheduler = self.cluster.scheduler
+        if self.cluster:
+            if hasattr(self.cluster, "scheduler") and self.cluster.scheduler:
+                scheduler = self.cluster.scheduler
+                if (
+                    hasattr(self.cluster, "scheduler_comm")
+                    and self.cluster.scheduler_comm
+                ):
+                    info = self.sync(self.cluster.scheduler_comm.identity)
+                else:
+                    info = self.cluster.scheduler.identity()
```

Comment:
So, I think that this is worse if we have a local Scheduler, and we're in async mode.  In that case `info` is a future, rather than a dictionary.

As an alternative, I recommend the following:

```diff
diff --git a/distributed/client.py b/distributed/client.py
index 381516da..01b1248a 100644
--- a/distributed/client.py
+++ b/distributed/client.py
@@ -786,10 +786,11 @@ class Client(Node):
             return "<%s: not connected>" % (self.__class__.__name__,)

     def _repr_html_(self):
+        from .scheduler import Scheduler
         if (
             self.cluster
             and hasattr(self.cluster, "scheduler")
-            and self.cluster.scheduler
+            and isinstance(self.cluster.scheduler, Scheduler)
         ):
             info = self.cluster.scheduler.identity()
             scheduler = self.cluster.scheduler
```

It's nice to use the local scheduler's identify function directly, because this doesn't require any async communication.  This makes it trivial to use whether or not we're in async mode.

Reply:
Is that change instead of this, or as well as? We

Reply:
Instead of, I think.  (I haven't tested it)

My guess is that the implementation currently in this PR would fail in Jupyter if we had code like the following:

```python
from dask.distributed import Client

client = await Client(asynchronous=True)
client
```